### PR TITLE
Add `\reference ProgramNames` to PythonPlugin:Instance

### PR DESCRIPTION
**Pull request by jmarrec**
Pull request overview
---------------------

Add `\reference ProgramNames` to PythonPlugin:Instance

cf https://github.com/NREL/OpenStudio/issues/4353#issuecomment-860572646 where it was first reported. I've made the change to OpenStudio's `ProposedEnergy+.idd` which does enforce it, but I'm propagating upstream here for coherence.

### Pull Request Author
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [ ] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [ ] Label the PR with at least one of: Defect, Refactoring, NewFeature, Performance, and/or DoNoPublish
 - [ ] Pull requests that impact EnergyPlus code must also include unit tests to cover enhancement or defect repair
 - [ ] Author should provide a "walkthrough" of relevant code changes using a GitHub code review comment process
 - [ ] If any diffs are expected, author must demonstrate they are justified using plots and descriptions
 - [ ] If changes fix a defect, the fix should be demonstrated in plots and descriptions
 - [ ] If any defect files are updated to a more recent version, upload new versions here or on DevSupport
 - [ ] If IDD requires transition, transition source, rules, ExpandObjects, and IDFs must be updated, and add IDDChange label
 - [ ] If structural output changes, add to output rules file and add OutputChange label
 - [ ] If adding/removing any LaTeX docs or figures, update that document's CMakeLists file dependencies

### Reviewer
This will not be exhaustively relevant to every PR.
 - [ ] Perform a Code Review on GitHub
 - [ ] If branch is behind develop, merge develop and build locally to check for side effects of the merge
 - [ ] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] If feature, test running new feature, try creative ways to break it
 - [ ] CI status: all green or justified
 - [ ] Check that performance is not impacted (CI Linux results include performance check)
 - [ ] Run Unit Test(s) locally
 - [ ] Check any new function arguments for performance impacts
 - [ ] Verify IDF naming conventions and styles, memos and notes and defaults
 - [ ] If new idf included, locally check the err file and other outputs

### CODE REVIEW

File: idd/Energy+.idd.in [hunk 1]
```diff
@@ -103363,6 +103363,7 @@ PythonPlugin:Instance,
    A1, \field Name
        \required-field
        \type alpha
+       \reference ProgramNames
```
Comment (jmarrec) [inline idd/Energy+.idd.in:103366]: That's it.

**Comment by mitchute** [comment]
I'm not sure what the purpose of the "Main Model Program Calling Manager" field. The model can definitely be run without having a main calling manager. Does that field need to be eliminated?

**Comment by mitchute** [comment]
Should we also be adding ```\reference ProgramNames``` to the "Plant Connection <#> Initialization Program Calling Manager" and "Plant Connection <#> Simulation Program Calling Manager" fields since those also accept plugin instances?

**Comment by mitchute** [comment]
Sorry, I got my wires crossed on this. Deleting my previous comments. 

This is good to go.

**Review by bonnema — COMMENTED** [review]
Looks good to me. Thank you!

**Pull request merged**